pkg/ruler/ruler.go

```diff
@@ -283,14 +284,14 @@ func (r *Ruler) newGroup(ctx context.Context, userID string, item *workItem) (*r
 	opts := &rules.ManagerOptions{
 		Appendable:  appendable,
 		QueryFunc:   rules.EngineQueryFunc(r.engine, r.queryable),
-		Context:     ctx,
+		Context:     context.Background(),
```

Comment:
why?

Reply:
Since this is potentially the only time we create the group now, I think we should use a context that will last for the entire lifecycle of the group. However, I don't think this context is actually used anywhere in the prometheus code since we do not use a prometheus manager.

Reply:
I'm not seeing why you think it's ok to give `ctx` to the `appendableAppender` but not to use it here.  The lifecycles seem the same to me.

Reply:
OK, if you don't have `ctx` then `context.Background()` seems fine.  What would `appendableAppender` be in this function - `nil` ?